app/code/core/Mage/Catalog/Model/Product.php

```diff
@@ -2312,4 +2317,18 @@ public function getEvent()
 
         return $event;
     }
+
+    /**
+     * @param int $storeId
+     * @return Mage_Review_Model_Review_Summary
+     */
+    public function getReviewSummary($storeId = null)
+    {
+        if (empty($this->_reviewSummary)) {
+            $this->_reviewSummary = Mage::getModel('review/review_summary')
+                ->setStoreId($storeId ?? Mage::app()->getStore()->getId())
+                ->load($this->getId());
+        }
+        return $this->_reviewSummary;
+    }
```

Comment:
It makes more sense for me to not touch `getEntitySummary` in `app/code/core/Mage/Review/Model/Review.php`, and if we set `$_reviewSummary = []`, then we can do this:
```php
    public function getReviewSummary($storeId = 0)
    {
       if (!array_key_exists($storeId, $this->_reviewSummary)) {
            Mage::getModel('review/review')->getEntitySummary($this, $storeId);
            $this->_reviewSummary[$storeId] = $this->getRatingSummary();
       }
    }
```

I didn't test the above code.

Reply:
:+1: